euphoria-core/src/main/java/cz/seznam/euphoria/core/client/dataset/windowing/GlobalWindowing.java

```diff
@@ -50,27 +53,27 @@ private Object readResolve() throws ObjectStreamException {
     }
 
     @Override
-    public int compareTo(BatchWindow o) {
+    public int compareTo(Window o) {
       return 0;
     }
   } // ~ end of Label
 
-  private final static Batch<?> INSTANCE = new Batch<>();
-  private Batch() {}
+  private final static GlobalWindowing<?> INSTANCE = new GlobalWindowing<>();
+  private GlobalWindowing() {}
 
   @Override
-  public Iterable<BatchWindow> assignWindowsToElement(WindowedElement<?, T> el) {
-    return Collections.singleton(BatchWindow.INSTANCE);
+  public Iterable<Window> assignWindowsToElement(WindowedElement<?, T> el) {
+    return Collections.singleton(Window.INSTANCE);
```

Comment:
Here is possible a micro optimization - we could create the set and store it in static field. It should be immutable, so it seems safe.